service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -190,56 +212,60 @@ private JsonObject getRequestPayload(final JsonObject request) {
         return payload;
     }
 
+    /**
+     * Check the request payload for validity.
+     * @param request The request that was sent via the vert.x event bus containing the payload to be checked.
+     * @return boolean The result of the check : {@link Boolean#TRUE} if the payload is valid, {@link Boolean#FALSE} otherwise.
+     * @throws ClassCastException If the {@link TenantConstants#FIELD_ADAPTERS_TYPE} element is not a {@link JsonArray}
```

Comment:
personally, I would expect an `IllegalArgumentException` in this case

Reply:
I agree - the vert.x json implementation throws the ClassCastException.
After attempted a version with `payload.getObject()` and checking for the type with `instanceof` operator (which lead to lengthy code that looked bad), I rather remap the ClassCastException to IllegalArgumentException.